appserver/ejb/ejb-opentracing/src/main/java/fish/payara/ejb/opentracing/OpenTracingIiopServerInterceptor.java

```diff
@@ -0,0 +1,156 @@
+
+/**
+ * IIOP Server Interceptor for propagating OpenTracing SpanContext to Payara Server.
+ *
+ * @author Andrew Pielage <andrew.pielage@payara.fish>
+ */
+public class OpenTracingIiopServerInterceptor extends LocalObject implements ServerRequestInterceptor {
+
+    private OpenTracingService openTracingService;
+
+    public OpenTracingIiopServerInterceptor(OpenTracingService openTracingService) {
+        this.openTracingService = openTracingService;
+    }
+
+    @Override
+    public void receive_request_service_contexts(ServerRequestInfo ri) throws ForwardRequest {
+
+    }
+
+    @Override
+    public void receive_request(ServerRequestInfo serverRequestInfo) throws ForwardRequest {
+        if (openTracingService == null) {
+            return;
+        }
+
+        Tracer tracer = openTracingService.getTracer(PAYARA_CORBA_RMI_TRACER_NAME);
+        if (tracer == null) {
+            return;
+        }
+
+        ServiceContext serviceContext = serverRequestInfo.get_request_service_context(OPENTRACING_IIOP_ID);
+        if (serviceContext == null) {
+            return;
+        }
+
+        ByteArrayInputStream bis = new ByteArrayInputStream(serviceContext.context_data);
+        OpenTracingIiopTextMap openTracingIiopTextMap = null;
+        try (ObjectInput in = new OpenTracingIiopObjectInputStream(bis)) {
+            openTracingIiopTextMap = (OpenTracingIiopTextMap) in.readObject();
+        } catch (IOException | ClassNotFoundException exception) {
+            throw new ForwardRequest(exception.getMessage(), serverRequestInfo);
+        }
+
+        Tracer.SpanBuilder spanBuilder = tracer.buildSpan("rmi")
+                .withTag(Tags.COMPONENT.getKey(), "ejb");
+
+        if (openTracingIiopTextMap != null) {
+            SpanContext spanContext = tracer.extract(Format.Builtin.TEXT_MAP, openTracingIiopTextMap);
+
+            // Add the propagated span as a parent
+            spanBuilder.asChildOf(spanContext);
+        }
+
+        // Start the span and mark it as active
+        spanBuilder.startActive(true).span();
+    }
+
+    @Override
+    public void send_reply(ServerRequestInfo serverRequestInfo) {
+        closeScope();
+    }
+
+    @Override
+    public void send_exception(ServerRequestInfo serverRequestInfo) throws ForwardRequest {
+        closeScope();
+    }
+
+    @Override
+    public void send_other(ServerRequestInfo serverRequestInfo) throws ForwardRequest {
+        closeScope();
+    }
+
+    private void closeScope() {
+        if (openTracingService == null) {
+            return;
+        }
+        Tracer tracer = openTracingService.getTracer(PAYARA_CORBA_RMI_TRACER_NAME);
+        if (tracer == null) {
+            return;
+        }
+
+        // Make sure active scope is closed - this is an entry point to the server so the currently active span
+        // **should** be the one started in receive_request
+        try (Scope activeScope = tracer.scopeManager().active()) {
```

Comment:
nothing happens with `activeScope`, seems wrong...

Reply:
It closes it (and the span it references too), scope is `Closeable`.